README.md

```diff
@@ -1,2 +1,344 @@
+    tbl.string('password', 128);
+  });
+};
+
+exports.down = function(knex, Promise) {
+  return knex.schema.table('users', (tbl) => {
+    tbl.dropColumn('password');
+  });
+};
+```
+
+In order to run this migration we can either wait until we run our server at which point the server's up function will run all the latest migrations, or we can do it manually now by doing `knex migrate:latest`.
+
+Now we have a password field in our database, but we still need to link the change to our Bookshelf model definition for user. To do this open the `app/models/user.js` file. Our first step here is to install the bcrypt module that we will use to encrypt the password. Do that by installing it with `npm install bcrypt --save`, and then require it at the top of the file like so: 
+
+```
+const bcrypt = require('bcrypt');
+```
+
+Now, finally, modify the User model definition so that it looks like the following:
+
+```
+const User = bookshelf.Model.extend({
+  tableName: 'users',
+  initialize: function() {
+    this.on('creating', this.encryptPassword);
+  },
+  hasTimestamps: true,
+  posts: function() {
+    return this.hasMany(Posts, 'author');
+  },
+  comments: function() {
+    return this.hasMany(Comments);
+  },
+  encryptPassword:(model, attrs, options) => {
+    return new Promise((resolve, reject) => {
+      bcrypt.hash(model.attributes.password, 10, (err, hash) => {
+        if (err) return reject(err);
+        model.set('password', hash);
+        resolve(hash);
+      });
+    });
+  },
+  validatePassword: function(suppliedPassword) {
+    let self = this;
+    return new Promise(function(resolve, reject) {
+      const hash = self.attributes.password;
+      bcrypt.compare(suppliedPassword, hash, (err, res) => {
+        if (err) return reject(err);
+        return resolve(res);
+      });
+    });
+  }
+});
+```
+
+So what is going on in this code? The first thing we're doing is adding an "override" for the User model's default `initalize` method, and within that override we are setting an event listener on the event called "creating". In other words, when the model is "creating" a new user we want it to call the specified method: `this.encryptPassword`. 
+
+Now, `encryptPassword`, as you can see, is a method that we've defined on the User model. That function returns a Promise that encrypts the user's supplied password using the bcrypt module. The promise either resolves with the hashed value of the password, or rejects, providing the error supplied by bcrypt as the reason.
+
+To check that this is working, add a new user by posting the appopriate data (i.e. name, username, email, password) to the `/user` route using either curl or Postman. Once you've created the user, check your database's users table. You should see a hash value in the password field that looks something like this:
+
+```
+$2a$10$pLOHxDVtdYQgemM2yVN.bOMTvWeMfRTV1ORgIlPPD0X9PBYHmPkCK 
+```
+
+Great, we're done configuring our schema.
```

Comment:
Could we include a paragraph here about why we don't ever want to store a plaintext password?